h_program-lang/Parse_info.ml

```diff
@@ -718,6 +718,21 @@ let adjust_pinfo_wrt_base base_loc loc =
         loc.column;
     file    = base_loc.file; }
 
+(* Token locations are supposed to denote the beginning of a token.
```

Comment:
I don't think it's a good idea because now when you get a token_location, you have no idea whether the line/col in it
represent the beginning or the end of the token. We could add another field in the record like StartToken or EndToken
but I think it's cleaner to make a function call line_col_charpos_at_end_of_token_location that returns
a triple (line x col x charpos) and let the caller use that information.

Reply:
Yeah, good call. I was going along with this because we already produce `token_location` values which correspond to the end of the token in our current code, but that's no reason to continue doing so.